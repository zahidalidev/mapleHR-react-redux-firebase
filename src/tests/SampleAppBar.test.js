import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import AppBar from 'components/sample/AppBar'

describe('Button', () => {
  test('renders', () => {
    expect(shallow(<AppBar />).exists()).toBe(true)
  })

  test('should render button', () => {
    const tree = renderer.create(<AppBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should have Model component', () => {
    const wrapper = shallow(<AppBar />)
    const modal = wrapper.find('ForwardRef(Modal)')
    expect(modal.exists()).toBe(true)
  })

  test('should have switch', () => {
    const wrapper = shallow(<AppBar />)
    const switchType = wrapper.find('ForwardRef(FormGroup)')
    expect(switchType.exists()).toBe(true)
  })

  test('should have Box', () => {
    const wrapper = shallow(<AppBar />)
    const box = wrapper.find('ForwardRef(Box)')
    expect(box.exists()).toBe(true)
  })

  test('default switch value should be true', () => {
    const wrapper = shallow(<AppBar />)
    const formControl = wrapper.find('ForwardRef(FormControlLabel)')
    expect(formControl.getElement().props.control.props.defaultChecked).toBe(true)
  })

  test('default color theme should be dark', () => {
    const wrapper = shallow(<AppBar />)
    const formControl = wrapper.find('div').first()
    expect(formControl.getElement().props.style.backgroundColor).toBe('#fff')
  })

  test('after clicking the switch the theme color should be light', () => {
    const wrapper = shallow(<AppBar />)
    wrapper.find('ForwardRef(FormControlLabel)').simulate('change')

    const formControl = wrapper.find('div').first()
    expect(formControl.getElement().props.style.backgroundColor).toBe('#000')
  })
})
