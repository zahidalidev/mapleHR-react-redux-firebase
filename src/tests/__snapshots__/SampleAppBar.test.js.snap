// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
<div
  className="d-flex row justify-content-center align-align-items-center"
  style={
    Object {
      "backgroundColor": "#fff",
      "height": 70,
      "width": "100%",
    }
  }
>
  <div>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Open modal
    </button>
  </div>
</div>
`;
